src/simulators/statevector/transformer.hpp

```diff
@@ -23,51 +23,50 @@ namespace QV {
 
 template <typename T> using cvector_t = std::vector<std::complex<T>>;
 
-template <typename data_t = double> class Transformer {
+template <typename Container, typename data_t = double> class Transformer {
 
   // TODO: This class should have the indexes.hpp moved inside it
 
 public:
+
+  virtual ~Transformer() {}
   //-----------------------------------------------------------------------
   // Apply Matrices
   //-----------------------------------------------------------------------
 
   // Apply a N-qubit matrix to the state vector.
   // The matrix is input as vector of the column-major vectorized N-qubit
   // matrix.
-  template <typename Container>
-  static void apply_matrix(Container &data, size_t data_size, int threads,
+
+  virtual void apply_matrix(Container &data, size_t data_size, int threads,
                            const reg_t &qubits, const cvector_t<double> &mat);
```

Comment:
If these methods are no-longer static, we can make them const:
```suggestion
                           const reg_t &qubits, const cvector_t<double> &mat) const;
```